### PR TITLE
UIIN 1321 Allow remote storage locations to be changed to non-remote storage location

### PR DESCRIPTION
**Pull request by axelhunn**
https://issues.folio.org/browse/UIIN-1321
Purpose: to allow a user to change an item or holding location from a remote storage location to a non-remote storage location

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/RemoteStorage/Provider.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import PropTypes from 'prop-types';
+
+import { stripesConnect } from '@folio/stripes/core';
+
+const Context = React.createContext({});
+
+export const useRemoteStorageApi = () => React.useContext(Context);
+
+const Provider = ({ resources, mutator, ...rest }) => {
+  const [persistentMutator] = React.useState(mutator);
```
Comment (aliaksei-chumakou) [inline src/RemoteStorage/Provider.js:11]: FYI, mutator is mutable

**Review by aliaksei-chumakou — APPROVED** [review]

**Review by uladislausamets — APPROVED** [review]

File: src/RemoteStorage/Provider.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  React.useEffect(() => {
```
Comment (NikitaSedyx) [inline src/RemoteStorage/Provider.js:13]: why do you access react utils via React? looks like it's done only by you

Comment (axelhunn) [reply src/RemoteStorage/Provider.js:13]: Kent C. Dodds taught me bad
https://kentcdodds.com/blog/useeffect-vs-uselayouteffect

Comment (NikitaSedyx) [reply src/RemoteStorage/Provider.js:13]: that's why you don't follow project codestyle?)

**Pull request merged**